pkgs/development/compilers/microscheme/default.nix

```diff
@@ -1,21 +1,24 @@
-{ lib, stdenv, fetchzip, vim, makeWrapper }:
+{ lib, stdenv, fetchFromGitHub, makeWrapper, vim }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation {
   pname = "microscheme";
   version = "0.9.3";
 
-  src = fetchzip {
-    name = "${pname}-${version}-src";
-    url = "https://github.com/ryansuchocki/microscheme/archive/v${version}.tar.gz";
-    sha256 = "1r3ng4pw1s9yy1h5rafra1rq19d3vmb5pzbpcz1913wz22qdd976";
+  src = fetchFromGitHub {
+    owner = "ryansuchocki";
+    repo = "microscheme";
+    rev = "b80901f4c7f42a372d51d92fe501263f6f54fd1a";
+    sha256 = "5qTWsBCfj5DCZ3f9W1bdo6WAc1DZqVxg8D7pwC95duQ=";
   };
 
-  buildInputs = [ makeWrapper vim ];
-
-  installPhase = ''
-    make install PREFIX=$out
+  postPatch = ''
+    substituteInPlace makefile --replace gcc ${stdenv.cc.targetPrefix}cc
   '';
 
+  nativeBuildInputs = [ makeWrapper vim ];
```

Comment:
You are sure about vim as a build time dependency?

Reply:
Yeah the build script calls `xxd`. Is there a more minimal way to provide this?

Reply:
I think unixtools.xxd should be the right one.

Reply:
Yeah, https://github.com/NixOS/nixpkgs/blob/0b7f4e404c23a3f4526166d029e3c59813721fbf/pkgs/top-level/unixtools.nix#L175